File: typescript/agentic-runtime/packages/agent-runtime/test/fixtures.ts
/**
 * Source: agentic-native-stack.md
 * Context: QN.2 Mock Provider Fixture
 * Extraction ID: CODE-106
 * Knowledge Links: KI-168
 * Status: scaffolded
 */

export class MockProvider implements Provider {
  private index = 0;

  constructor(private responses: ModelResponse[]) {}

  async complete(_request: ModelRequest): Promise<ModelResponse> {
    const response = this.responses[this.index % this.responses.length];
    this.index += 1;
    return response;
  }
}

File: rust/agentic-core/crates/agentic-daemon/tests/fixtures.rs
//! Source: agentic-native-stack.md
//! Context: QN.3 Temporary Workspace Fixture
//! Extraction ID: CODE-107
//! Knowledge Links: KI-168
//! Status: scaffolded

pub struct TempWorkspace {
    path: PathBuf,
}

impl TempWorkspace {
    pub async fn from_template(template: &Path) -> Result<Self, TestError> {
        let path = std::env::temp_dir().join(format!(
            "agentic-test-{}",
            uuid::Uuid::now_v7()
        ));

        copy_recursively(template, &path).await?;

        Ok(Self { path })
    }

    pub fn path(&self) -> &Path {
        &self.path
    }
}

impl Drop for TempWorkspace {
    fn drop(&mut self) {
        let _ = std::fs::remove_dir_all(&self.path);
    }
}